File: app/src/main/java/com/s15814/healthApp/BmiCalculator.java
package com.s15814.healthApp;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.navigation.fragment.NavHostFragment;

import java.text.DecimalFormat;

public class BmiCalculator extends Fragment {

    EditText heightInput, weightInput;
    TextView result;

    @Override
    public View onCreateView(
            LayoutInflater inflater, ViewGroup container,
            Bundle savedInstanceState
    ) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.bmi_calculator, container, false);
    }

    public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        heightInput = (EditText) view.findViewById(R.id.insertHeight);
        weightInput = (EditText) view.findViewById(R.id.insertHeight);
        result = (TextView) view.findViewById(R.id.bmiCalculatorResultView);

        view.findViewById(R.id.bmiCalculatorCalculateButton).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                calculateBMI();
            }
        });

        view.findViewById(R.id.bmiCalculatorBackButton).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                NavHostFragment.findNavController(BmiCalculator.this)
                        .navigate(R.id.action_FirstFragment_to_mainMenu);
            }
        });
    }

    private void calculateBMI() {
        String heightStr = heightInput.getText().toString();
        String weightStr = weightInput.getText().toString();

        DecimalFormat df = new DecimalFormat("##.##");

        if (!heightStr.isEmpty() && !weightStr.isEmpty() && !heightStr.equals("0")) {
            double heightValue = Double.parseDouble(heightStr) / 100.00;
            double weightValue = Double.parseDouble(weightStr);

            double bmiValue = weightValue / Math.pow(heightValue, 2);
            double formattedBmi = Double.parseDouble(df.format(bmiValue));

            displayResult(formattedBmi);
        } else {
            result.setText(R.string.incorrectValueText);
        }
    }

    private void displayResult(double bmiValue) {
        String bmiResult;

        if (bmiValue < 18.50) {
            bmiResult = "Underweight";
        } else if (bmiValue < 24.90) {
            bmiResult = "Healthy";
        } else if (bmiValue < 29.90) {
            bmiResult = "Overweight";
        } else {
            bmiResult = "Obese";
        }

        bmiResult = bmiValue + "\nWeight status: " + bmiResult;
        if (bmiValue > 0.00) {
            result.setText(bmiResult);
        } else {
            result.setText(R.string.invalidInputText);
        }
    }

}

File: app/src/main/java/com/s15814/healthApp/BmiCalculator.kt
package com.s15814.healthApp

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.EditText
import android.widget.TextView
import androidx.fragment.app.Fragment
import androidx.navigation.fragment.NavHostFragment
import java.text.DecimalFormat
import kotlin.math.pow

class BmiCalculator : Fragment() {
    private var heightInput: EditText? = null
    private var weightInput: EditText? = null
    private var result: TextView? = null
    override fun onCreateView(
            inflater: LayoutInflater, container: ViewGroup?,
            savedInstanceState: Bundle?
    ): View? { // Inflate the layout for this fragment
        return inflater.inflate(R.layout.bmi_calculator, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        heightInput = view.findViewById<View>(R.id.insertHeight) as EditText
        weightInput = view.findViewById<View>(R.id.insertHeight) as EditText
        result = view.findViewById<View>(R.id.bmiCalculatorResultView) as TextView
        view.findViewById<View>(R.id.bmiCalculatorCalculateButton).setOnClickListener { calculateBMI() }
        view.findViewById<View>(R.id.bmiCalculatorBackButton).setOnClickListener {
            NavHostFragment.findNavController(this@BmiCalculator)
                    .navigate(R.id.action_FirstFragment_to_mainMenu)
        }
    }

    private fun calculateBMI() {
        val heightStr = heightInput!!.text.toString()
        val weightStr = weightInput!!.text.toString()
        val df = DecimalFormat("##.##")
        if (heightStr.isNotEmpty() && weightStr.isNotEmpty() && heightStr != "0") {
            val heightValue = heightStr.toDouble() / 100.00
            val weightValue = weightStr.toDouble()
            val bmiValue = weightValue / heightValue.pow(2.0)
            val formattedBmi = df.format(bmiValue).toDouble()
            displayResult(formattedBmi)
        } else {
            result!!.setText(R.string.incorrectValueText)
        }
    }

    private fun displayResult(bmiValue: Double) {
        var bmiResult: String = when {
            bmiValue < 18.50 -> {
                "Underweight"
            }
            bmiValue < 24.90 -> {
                "Healthy"
            }
            bmiValue < 29.90 -> {
                "Overweight"
            }
            else -> {
                "Obese"
            }
        }
        bmiResult = "$bmiValue\nWeight status: $bmiResult"
        if (bmiValue > 0.00) {
            result!!.text = bmiResult
        } else {
            result!!.setText(R.string.invalidInputText)
        }
    }
}